src/polkadot/PolkadotEngine.ts

```diff
@@ -210,6 +245,217 @@ export class PolkadotEngine extends CurrencyEngine<
+            if (
+              this.findTransaction(
+                this.currencyInfo.currencyCode,
+                edgeTransaction.txid
+              ) !== -1
+            ) {
+              hasNextPage = false
+              break
+            }
+
+            this.addTransaction(this.currencyInfo.currencyCode, edgeTransaction)
+          }
+        }
+
+        // Update progress
+        const totalCount = cleanTransfersResponse.data.transfers.totalCount
+        processedCount += transfers.length
+
+        this.tokenCheckTransactionsStatus[this.currencyInfo.currencyCode] =
+          totalCount === 0 ? 1 : processedCount / totalCount
+        this.updateOnAddressesChecked()
+      } catch (e: any) {
+        this.warn(`Error fetching Liberland transactions: ${e.message}`)
+        throw e
+      }
+    }
+
+    this.tokenCheckTransactionsStatus[this.currencyInfo.currencyCode] = 1
+    this.updateOnAddressesChecked()
+
+    // LLM Transfers. Endpoint only handles specifically the LLM token
+    hasNextPage = true
+    endCursor = null
+    const tokenId = '1'
+    const currencyCode = this.allTokensMap[tokenId].currencyCode
+
+    while (hasNextPage) {
+      const meritsOperationName = 'Merits'
+      const meritsQuery = `query Merits($userId: String, $cursor: Cursor) {  
+        merits(
+        filter: {or: [{fromId: {equalTo: $userId}}, {toId: {equalTo: $userId}}]}
+        after: $cursor
+        first: 50
+        orderBy: BLOCK_NUMBER_DESC
+        ) {
+          nodes {
+            id
+            fromId
+            toId
+            value
+            eventIndex
+            blockId
+            block {
+              number
+              timestamp
+            }
+          }
+          pageInfo {
+            hasNextPage
+            endCursor
+          }
+          totalCount
+        }
+      }`
+
+      try {
+        const meritsResponses = await this.fetchLiberlandScan(
+          endpoint,
+          meritsOperationName,
+          meritsQuery,
+          {
+            userId,
+            cursor: endCursor
+          }
+        )
+        const cleanMeritsResponse = asLiberlandMeritsResponse(meritsResponses)
+
+        const totalCount = cleanMeritsResponse.data.merits.totalCount
+        const meritTransfers = cleanMeritsResponse.data.merits.nodes
+        hasNextPage = cleanMeritsResponse.data.merits.pageInfo.hasNextPage
+        endCursor = cleanMeritsResponse.data.merits.pageInfo.endCursor
+
+        for (const tx of meritTransfers) {
+          const edgeTransaction = processLiberlandTransaction(
+            {
+              walletId: this.walletId,
+              walletInfo: this.walletInfo,
+              currencyInfo: this.currencyInfo,
+              allTokensMap: this.allTokensMap,
+              tokenId
+            },
+            tx
+          )
+          if (edgeTransaction != null) {
+            if (
+              this.findTransaction(currencyCode, edgeTransaction.txid) !== -1
```

Comment:
For whatever reasons this isn't exiting early on the first loop which means the entire history of the wallet is queried every time the user logs in. Save the latest txid as a variable on otherData (lots of other engines do this and you'll need to track a txid each for lld and llm) and exit early on that.

Reply:
Cause it wasn't normalized. Even then, searching this way makes this break on an unconfirmed tx. Though the engine never saves unconfirmed tx's, the GUI does immediately after sending, so in that situation the send tx never gets updated from pending.

Updated to check confirmation state.